aptos-move/block-executor/src/executor.rs

```diff
@@ -264,6 +274,7 @@ where
             }
 
             if last_input_output.execution_error(txn_idx).is_some() {
+                let mut maybe_error = maybe_error.lock();
```

Comment:
Are we now overwriting the errors? The flow isn't very clear?
Also can we call halt twice? what will happen in that case?

Reply:
The variable might. Tho the final result will be the same. The early exit can only halt with Abort (not skip rest)

Reply:
Ok, so I missed return (let's do else as you suggested), so higher abort can't overwrite, should be fine now but let me think some more.